### PR TITLE
JDK-8298255: JFR provide information about dynamization of number of compiler threads

### PR DESCRIPTION
**Pull request by MBaesken**
The JVM supports both dynamic GC threads, and dynamic Compiler threads.
While the GCConfiguration JFR event (see https://sap.github.io/SapMachine/jfrevents/#gcconfiguration ) seems to have at least some info about dynamization of GC threads ( field usesDynamicGCThreads), the JIT CompilerConfiguration event seems to miss this info (about JIT compiler threads dynamization), this should be added; currently the impression is given that the number is static but this is often not the case.
As an extension, the dynamization process of adding / removing threads could be added as well to the JFR events but I am not sure if this is desired so I just started with a very simple patch.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8298255](https://bugs.openjdk.org/browse/JDK-8298255): JFR provide information about dynamization of number of compiler threads

### Reviewers
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to [42ff5021](https://git.openjdk.org/jdk/pull/11563/files/42ff502113a6a68a23960b506cf9df06191143ae)
 * [Markus Grönlund](https://openjdk.org/census#mgronlun) (@mgronlun - **Reviewer**) ⚠️ Review applies to [42ff5021](https://git.openjdk.org/jdk/pull/11563/files/42ff502113a6a68a23960b506cf9df06191143ae)
 * [Erik Gahlin](https://openjdk.org/census#egahlin) (@egahlin - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11563/head:pull/11563` \
`$ git checkout pull/11563`

Update a local copy of the PR: \
`$ git checkout pull/11563` \
`$ git pull https://git.openjdk.org/jdk pull/11563/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11563`

View PR using the GUI difftool: \
`$ git pr show -t 11563`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11563.diff">https://git.openjdk.org/jdk/pull/11563.diff</a>

</details>

### CODE REVIEW

**Review by tstuefe — APPROVED** [review]
Seems reasonable, seeing that we do the same for GC. Good assuming tests are green.

**Review by mgronlun — APPROVED** [review]

**Comment by mgronlun** [comment]
@MBaesken Please put this back before UTC 16.00 today :)

**Comment by MBaesken** [comment]
Hi Markus and Thomas,  unfortunately I learned that the event field names must not be longer than 32 chars.
Otherwise  test  jdk/jfr/event/metadata/TestEventMetadata.java  fails with 

java.lang.RuntimeException: Name should not exceed 32 characters: expected true, was false
	at jdk.test.lib.Asserts.fail(Asserts.java:594)
	at jdk.test.lib.Asserts.assertTrue(Asserts.java:486)
	at jdk.jfr.event.metadata.TestEventMetadata.verifyName(TestEventMetadata.java:149)
	at jdk.jfr.event.metadata.TestEventMetadata.verifyValueDescriptor(TestEventMetadata.java:130)
	at jdk.jfr.event.metadata.TestEventMetadata.verifyValueDesscriptors(TestEventMetadata.java:125)
	at jdk.jfr.event.metadata.TestEventMetadata.main(TestEventMetadata.java:101)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:125)
	at java.base/java.lang.Thread.run(Thread.java:1623)

So I shortened the field name - are you still good with the latest version ?

File: test/jdk/jdk/jfr/event/compiler/TestCompilerConfig.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public static void main(String[] args) throws Exception {
             System.out.println("Event:" + event);
             Events.assertField(event, "threadCount").atLeast(0);
             Events.assertField(event, "tieredCompilation");
+            Events.assertField(event, "usesDynamicNrOfCompilerThreads");
```
[the remark this answers is not in the document]
Comment (MBaesken) [reply test/jdk/jdk/jfr/event/compiler/TestCompilerConfig.java:55]: would you be okay with  usesDynamicCompilerThreadCount  ? Sounds maybe a bit nicer .

[the remark this answers is not in the document]
Comment (egahlin) [reply test/jdk/jdk/jfr/event/compiler/TestCompilerConfig.java:55]: Perhaps "dynamicCompilerThreadCount". It's a boolean so use may not be needed.

**Comment by MBaesken** [comment]
Hi Erik,  I renamed the field to dynamicCompilerThreadCount.

**Comment by mgronlun** [comment]
Yep, good :)

**Pull request closed without merging**